docs/investigations/file-upload-with-graphql.md

```diff
@@ -0,0 +1,121 @@
+# File upload with graphql
+
+## Overview
+
+This document discuss the following:
+1. Issue of uploading files with graphql.
+2. Suggest and idea on handling large data objects.
+
+As vanilla graphql does not support uploading files, there are several workarounds:
+- Multipart Request Spec (implemented in gqlgen library)
+- Base64 data encoding
+- Middleware which handles file uploading
+
+### File uploading with gqlgen 0.9.0 library
```

Comment:
```suggestion
## Upload files with gqlgen 0.9.0 library
```

Reply:
After the heading and before the first step as an introductory sentence:
`To upload a file using gqlgen 0.9.0 library, follow these steps:`